verilog/analysis/checkers/struct_union_name_style_rule.cc

```diff
@@ -79,15 +82,65 @@ void StructUnionNameStyleRule::HandleSymbol(const verible::Symbol& symbol,
       // Neither a struct nor union definition
       return;
     }
-    const auto* identifier_leaf = GetIdentifierFromTypeDeclaration(symbol);
+    const auto *identifier_leaf = GetIdentifierFromTypeDeclaration(symbol);
     const auto name = ABSL_DIE_IF_NULL(identifier_leaf)->get().text();
-    if (!verible::IsLowerSnakeCaseWithDigits(name) ||
+    // get rid of the exceptions specified by user
+    std::string name_filtered;
+    if (!exceptions_.empty()) {
```

Comment:
I don't think we can do it this simple (see below about configuration).

We _do_ want to allow configurations that just contain units, such as `GiB`, and want to allow all items that looks like
`<some number>GiB`.

So instead of re-writing the name it is probably better to check this the other way around; this implementation is mostly a remnant from the earlier times when you had to iterate through the configuration string here thus things needed to be in order of that string, but now things are much easier as we have a `std::set` of exceptions, so we can instead look-up exceptions directly in the set. Which also makes it less brittle and much easier to understand (If I didn't know what you try to achieve, I would have to read through this code multiple times to understand what its intend is. Let's make it simpler for us looking at this code in the future...)

So rough implementation suggestion would probably be to go through the name in order, do early-out once we find one of the issues and use the capabilities of directly finding something in a `std::set` to test for the rest.
That way, things are easy to maintain in the future as each step is simple.

Pseudo-code of what I mean:
```
if (!absl::endsWith(name, "_t")) {
   violations.insert(...);
   return;  // detected already first violation: early out.
}
if (name[0] == '_') {
   violations.insert("can not start with underscore...");  // this allows us also to be more specific
   return;   // early out. violation detected, not a problem anymore for our later analysis
}
if (absl::IsLowerCaseSnakeWithDigits(...)) {
  return;  // Name looks cool already. We could leave this out as we essentially implement everyting below.
}
// now, our turn
for (const auto &part : absl::Split(name, '_')) {
   if (alllowercaseordigits(part)) continue;
   if (!isdigit(*part.begin()) {
       violation("sections with unit names need to start with digit");
       return;
   }
   if (exceptions.find(part) != exceptions.end()) continue;  // direct match of things that have digits in exceptions, e.g. 12B
   if (exceptions.find(remove_leading_digits(part)) == exceptions.end()) {
       violation("found digit followed by unit that is not configured as allowed exception");
       return;
   }
}
```